### ISSUE TITLE
CORS Error suddenly on all API calls

### ISSUE THREAD

**Issue by mwdwrd** [ProjectOpenSea/opensea-js#79]
Suddenly getting CORS Errors when calling `createfactorySellOrder` on a nextjs/vercel frontend.
Worked for a week, didn't change anything, and now doesn't work.

Has anything changed on your server to not allow these requests?

![image](https://user-images.githubusercontent.com/5651272/113214926-26baa980-9248-11eb-9db7-f9fe1cac5ae5.png)

**Comment by mwdwrd** [comment]
Localhost works fine.. It's when its on a server that's the problem.

**Comment by alexanderatallah** [comment]
Oops, we changed a bunch of CORS settings on our end and our tests failed to catch a corner case - fixing now! should be back to normal in 15 min

**Comment by mwdwrd** [comment]
Thansk @alexanderatallah – Much appreciated! I can finally get some sleep 💯

**Comment by NielsDom** [comment]
@mwdwrd is it indeed working on localhost but on a public server. It is not possible to use it on a public server? 

Thanks

**Comment by mwdwrd** [comment]
Can confirm now working @alexanderatallah. Thanks for the quick responses.

**Comment by Iroh-Omolola** [comment]
@alexanderatallah I'm having issue trying to use the opensea test to access my collection. 

Access to XMLHttpRequest at 'https://testnets-api.opensea.io/assets?asset_contract_address=0xF42029DD93A9B20805152b804114380d6acba959&order_direction=asc' from origin 'http://localhost:3001' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
xhr.js:210 
        
       GET https://testnets-api.opensea.io/assets?asset_contract_address=0xF42029DD93A9B20805152b804114380d6acba959&order_direction=asc net::ERR_FAILED 200
dispatchXhrRequest @ xhr.js:210
xhrAdapter @ xhr.js:15
dispatchRequest @ dispatchRequest.js:58
request @ Axios.js:108
Axios.<computed> @ Axios.js:129
wrap @ bind.js:9
getMyNfts @ App.js:11
(anonymous) @ App.js:17
invokePassiveEffectCreate @ react-dom.development.js:23487
callCallback @ react-dom.development.js:3945
invokeGuardedCallbackDev @ react-dom.development.js:3994
invokeGuardedCallback @ react-dom.development.js:4056
flushPassiveEffectsImpl @ react-dom.development.js:23574
unstable_runWithPriority @ scheduler.development.js:468
runWithPriority$1 @ react-dom.development.js:11276
flushPassiveEffects @ react-dom.development.js:23447
(anonymous) @ react-dom.development.js:23324
workLoop @ scheduler.development.js:417
flushWork @ scheduler.development.js:390
performWorkUntilDeadline @ scheduler.development.js:157
createError.js:16 
        
       Uncaught (in promise) Error: Network Error
    at createError (createError.js:16)
    at XMLHttpRequest.handleError (xhr.js:117)

**Comment by paik1** [comment]
Hi Im getting a similar error as well
Access to XMLHttpRequest at 'https://testnets-api.opensea.io/assets?asset_contract_address=0x18dA093DC8a3288DdA3A3c3A18973eA17cb48943' from origin 'https://paik1.github.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by HomemadeToast57** [comment]
Having issue Access to fetch at 'https://testnets-api.opensea.io/assets?asset_contract_address=0xdadd397c13b5a74566E939754CAaFd6E8aF7eCd0&order_direction=asc' from origin 'https://61d7c46bd5dcaf94494ca0a4--nft-toast.netlify.app' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

**Comment by Haugen** [comment]
I'm getting the same issue using api.opensea.io. Making API calls works fine on localhost, but as soon as I deploy to a server I get the same `has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.` error. What are we missing here?

**Comment by HomemadeToast57** [comment]
> I'm getting the same issue using api.opensea.io. Making API calls works fine on localhost, but as soon as I deploy to a server I get the same `has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.` error. What are we missing here?

I made a new issue that is currently open (this one is closed). Feel free to comment on that one. :)

**Comment by huyredfox** [comment]
I have the same issue today
Access to fetch at 'https://api.opensea.io/wyvern/v1/orders/post/' from origin 'https://mydomain.vercel.app' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

Please help @alexanderatallah

**Comment by laurafiuza** [comment]
Same for me!

**Comment by oveddan** [comment]
CORS issues happening again

**Comment by rapendivya** [comment]
please watchout for cors. it's really a pain.

**Comment by swdreams** [comment]
CORS issue happening again on API doc page. https://docs.opensea.io/reference/getting-assets

**Comment by stanleynguyen** [comment]
It's happening again :(

**Comment by traderkeo** [comment]
Man.. how hard is it for a billion dollar company to manage like 7 endpoints..

**Comment by afrotonder** [comment]
Having this same problem. Plz OpenSea add CORS policy for docs.opensea.io. 😭

**Comment by ElamuruganGR** [comment]
Access to XMLHttpRequest at 'https://testnets-api.opensea.io/assets?asset_contract_address=0xb94e6bF1248F9fF9eb15A6B74d7008170de61E0a&order_direction=asc' from origin 'https://hardcore-nobel-a232a2.netlify.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

I am also facing the CORS error.

**Comment by adamPatrick12** [comment]
CORS error is still present :(

**Comment by MagusVeles** [comment]
This issue started happening in our project as well.

I'm just dropping in to see if there is something we can about this or do we need to wait until OS fixes this?

**Comment by chrisco255** [comment]
This issue needs to be reopened.

**Comment by luquitared** [comment]
Also experiencing this issue still

**Comment by ochikov** [comment]
Same here:
```
'https://testnets-api.opensea.io/api/v1/collections?asset_owner=0x5cA5509027CD45381Bd2Ebca46B7e139533ae9AA&offset=0&limit=300' has been blocked by CORS policy: Request header field access-control-allow-origin is not allowed by Access-Control-Allow-Headers in preflight response.
```

**Comment by HbsAdithya** [comment]
Try to use this chrome extension "Allow CORS: Access-Control-Allow-Origin"

**Comment by polyglotguru1** [comment]
I'm two months into programming. I was told that web3 was all about decentralization. Nothing could be further from the truth. I thought to deploy my app like a normal person, but I see that there are forces in the web3 world such as opensea who would not have it so.

**Comment by iamDJB** [comment]
> Oops, we changed a bunch of CORS settings on our end and our tests failed to catch a corner case - fixing now! should be back to normal in 15 min

@alexanderatallah has this happened again? Cannot access due to CORS policy...

**Comment by Captain-Oski** [comment]
Same Cors problem, Yesterday worked fine. Now Cors Error

**Comment by montasaurus** [comment]
> Same Cors problem, Yesterday worked fine. Now Cors Error

On which domain? @Captain-Oski @ochikov
